### ISSUE TITLE
Pathmap does not seem to be working, resulting in determinism failure

### ISSUE THREAD

**Issue by jnm2** [dotnet/sdk#15321]
**Version Used**: .NET SDK 5.0.101

For simplicity the repro doesn't reference an NuGet package that includes .cs files out of the package cache, but /pathmap appears to not be working for .cs files out of the package cache either. I can provide a repro involving the package cache if that is useful.

##### Repro.csproj

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <IncludeSymbols>true</IncludeSymbols>
    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
  </PropertyGroup>

</Project>
```

##### Class1.cs

```cs
public class Class1 { }
```

`dotnet pack -c Release` creates a package that does this:

![image](https://user-images.githubusercontent.com/8040367/104078387-cdfce700-51eb-11eb-9285-de39e8960ba2.png)

Here's what the SDK provided to csc:
![image](https://user-images.githubusercontent.com/8040367/104078488-4794d500-51ec-11eb-8427-0d52b5d1e926.png)

Here are the filenames in the PDB (in the case of the repro that includes source from the package cache, the package cache path is not mapped either):
![image](https://user-images.githubusercontent.com/8040367/104078515-7ca12780-51ec-11eb-9266-290e7b4fb6fc.png)

**Comment by jnm2** [comment]
@jaredpar Yes. The reason I opened it here and not in the SDK repo is because the .targets file that creates the pathmap seems to live in the Roslyn repository as shown by https://github.com/dotnet/roslyn/pull/25325/files#diff-ff36db066709f0eb75570259566cf9a14c4009fb1d1ead1097b040c7ff27c3e4R124.

**Comment by jaredpar** [comment]
Hmm, @tmat what is the appropriate repo for this bug?

**Comment by jnm2** [comment]
> Not sure where the `;` convention comes from but it's not supported in the [code](http://sourceroslyn.io/#Microsoft.CodeAnalysis/CommandLine/CommandLineParser.cs,260) or in the [documentation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/pathmap-compiler-option)

@jaredpar The docs you linked currently say this:

> To specify multiple mapped source paths, separate each with a semicolon.

**Comment by jnm2** [comment]
The 5.0.302 SDK seems to have switched to using commas instead of semicolons in the pathmap. This bug was fixed somewhere between 5.0.101 and 5.0.302.

**Comment by tmat** [comment]
https://github.com/NuGet/Home/issues/9810
